DigitalLearningSolutions.Web/Helpers/CourseDetailsValidator.cs

```diff
@@ -0,0 +1,112 @@
+﻿namespace DigitalLearningSolutions.Web.Helpers
+{
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Extensions;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup.CourseDetails;
+    using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+    public static class CourseDetailsValidator
+    {
+        public static void ValidateCustomisationName(
+            EditCourseDetailsFormData formData,
+            ModelStateDictionary modelState,
+            ICourseService courseService,
+            int centreId,
+            int customisationId = 0
+        )
+        {
+            formData.ConvertCustomisationNameToEmptyStringIfNull();
+            if (string.IsNullOrEmpty(formData.CustomisationName) && courseService.DoesCourseNameExistAtCentre(
+                formData.CustomisationName!,
+                centreId,
+                formData.ApplicationId,
+                customisationId
+            ))
+            {
+                modelState.AddModelError(
+                    nameof(formData.CustomisationName),
+                    "A course with no add on already exists"
+                );
+            }
+            else if (string.IsNullOrEmpty(formData.CustomisationName))
+            {
+                modelState.ClearErrorsOnField(nameof(formData.CustomisationName));
+            }
+            else if (formData.CustomisationName.Length > 250)
+            {
+                modelState.AddModelError(
+                    nameof(formData.CustomisationName),
+                    "Course name must be 250 characters or fewer, including any additions"
+                );
+            }
+            else if (courseService.DoesCourseNameExistAtCentre(
+                formData.CustomisationName,
+                centreId,
+                formData.ApplicationId,
+                customisationId
+            ))
+            {
+                modelState.AddModelError(
+                    nameof(formData.CustomisationName),
+                    "Course name must be unique, including any additions"
+                );
+            }
+        }
+
+        public static void ValidatePassword(EditCourseDetailsFormData formData, ModelStateDictionary modelState)
+        {
+            if (formData.PasswordProtected)
+            {
+                return;
+            }
+
+            if (modelState.HasError(nameof(formData.Password)))
+            {
+                modelState.ClearErrorsOnField(nameof(formData.Password));
+            }
+
+            formData.Password = null;
+        }
+
+        public static void ValidateEmail(EditCourseDetailsFormData formData, ModelStateDictionary modelState)

```

Comment:
I don't think this method name fits the method purpose as it is `not` validating but resetting/clearing validation and also resetting the value.

Reply:
changed to ResetValueAndClearErrorsOnEmailIfUnselected